pkg/synthetictests/image_pulls.go

```diff
@@ -0,0 +1,48 @@
+package synthetictests
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	"github.com/openshift/origin/pkg/test/ginkgo/junitapi"
+)
+
+// testBackoffPullingRegistryRedhatImage looks for this symptom:
+//   reason/ContainerWait ... Back-off pulling image "registry.redhat.io/openshift4/ose-oauth-proxy:latest"
+//   reason/BackOff Back-off pulling image "registry.redhat.io/openshift4/ose-oauth-proxy:latest"
+// to happen over a certain threshold and if so, marks it as a flake so we can see how often it's
+// happening.
+//
+func testBackoffPullingRegistryRedhatImage(events monitorapi.Intervals) []*junitapi.JUnitTestCase {
```

Comment:
So this appears to match the other one, but could we do

```go
type eventRecognizerFunc func(event monitorapi.Interval) bool

type matchEventForRegexOrDie(regex string) eventRecognizerFunc{
  parse(regex)
  if err {
    panic
  }
  return func(){
    return regExp.MatchString(e.Message)
  }
}

type singleEventCheckRegex struct{
  testName string
  recognizer eventRecognizerFunc
  failThreshold int
}

func (s *singleEventCheckRegex) test(events) junittestcases{
	for _, e := range events {
		if s.recognizer(e) {

			// We found an instance of "Back-off pulling image"; if it happens more than some
			// threshold, report it as a flake.
			//
			msg := fmt.Sprintf("%s - %s", e.Locator, e.Message)
			eventDisplayMessage, times := getTimesAnEventHappened(msg)
			if times > s.failThreshold {
				output := fmt.Sprintf("event [%s] happened %d times\n", eventDisplayMessage, times)
				failure := &junitapi.JUnitTestCase{
					Name:      testName,
					SystemOut: output,
					FailureOutput: &junitapi.FailureOutput{
						Output: output,
					},
				}
				// Include a fake success so this will always be a "flake" for now.
				return []*junitapi.JUnitTestCase{failure, success}
			}
		}
	}

	return []*junitapi.JUnitTestCase{success}
}

func testBackoffPullingRegistryRedhatImage(events){
  testName := "[sig-arch] should not see excessive pull back-off on registry.redhat.io"   return 
  return newSingleEventCheckRegex(testName, regex).test(events)
}

```

Reply:
Then we can argue about a flakeThreshold (you'll need one to use here I think), and auto-exclude from the "normal" event test (this can come next week I think).

Reply:
whoa.  nice.  Thanks.  I added that with tweaks to get it to compile ok -- hopefully I interpreted what you are saying correctly (since it now seems more obvious how we can add another check using another regex with just a few lines).